File: leetcode57.js
/**
 * @param {number[][]} intervals
 * @param {number[]} newInterval
 * @return {number[][]}
 */
var insert = function(intervals, newInterval) {
    const res = []
    const n = intervals.length
    let last = newInterval[0]
    let cur = newInterval[1]
    flag = true
    for(let i = 0;i < n;++i)
    {   
        if(flag){
           if(last <= intervals[i][0] && cur >=intervals[i][0]){
                cur = Math.max(cur,intervals[i][1])
                continue
           }else if(last >= intervals[i][0] && last <= intervals[i][1]){
                cur = Math.max(cur,intervals[i][1])
                last = intervals[i][0]
                continue
           }
            if(cur < intervals[i][0]) {
                res.push([last,cur])
                flag = false
                --i
                continue
            }
        }
        res.push(intervals[i])
    }
    if(flag) res.push([last,cur])
    return res;
};

File: leetcode57.cpp
#include <vector>
#include <algorithm>
using namespace std;
class Solution {
    public:
        vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
            vector<vector<int>> res;
            vector<vector<int>>::iterator it = intervals.begin();
            while(it != intervals.end()){
                if((*it)[0] > newInterval[0]) break;
                ++it;
            }
            intervals.insert(it,newInterval);
            int last = intervals[0][0],cur = intervals[0][1];
            int n = intervals.size();           
            for(int i = 1;i < n;++i)
            {
                if(cur >= intervals[i][0] and cur <= intervals[i][1]){
                    cur = intervals[i][1];
                    continue;
                }else if(cur > intervals[i][1]) continue;
                res.push_back({last,cur});
                last = intervals[i][0],cur = intervals[i][1];
             
            }
            res.push_back({last,cur});
            return res;
        }
    };